editor/src/clj/editor/app_view.clj

```diff
@@ -790,13 +791,40 @@
                                 :text "If the engine is already running, shut down the process manually and retry"}]}
            :content (.getMessage e)})))))
 
+(defn- owning-resource [basis node-id]
+  (when-some [owning-resource-node-id (core/scope-of-type basis node-id resource-node/ResourceNode)]
+             [owning-resource (resource-node/resource basis owning-resource-node-id)]
+    owning-resource))
+
+(defn- get-cycle-detected-help-message
+  [node-type-name node-id]
+  (let [basis (g/now)
+        proj-path (some-> (owning-resource basis node-id) resource/proj-path)
+        resource-path (some? proj-path) proj-path "'unknown'"]
```

Comment:
I'm always getting "unknown" here. Not sure if I have adapted the initial code snippet from @matgis correctly?